examples/cfq/syntax-based/grammar.py

```diff
@@ -74,29 +77,98 @@ def generate_grammar(grammar_str: str):
     rules.append(current_rule)
   sub_rules = []
   for rule in rules:
-    sub_rules += generate_sub_rules(rule)
+    sub_rules += extract_sub_rules_from_rules(rule)
+  return sub_rules
+
+class TermType(Enum):
+  SYNTAX_TERM = 0
+  REGEX_TERM = 1
+  RULE_TERM = 2
+
+class Term:
+  """
+  Class describing a term (token on the right side of a rule). It contains a
```

Comment:
Here you say a Term describes a token on the right side of a rule, but in the end of the comment you say "if the term is a rule head ...". That seems to be contradicting?

Reply:
will be clarified in the comment